Reposition animated spending bucket over DC

rachelekm opened the pull request:
## Overview

Builds off of the work of #75 * and positions the animated spending bucket to be fixed over DC. In order for dynamic content to continue to re-render from within a leaflet UI layer as the time value (and associated spending values) updates, we take advantage of [createPortals](https://react.dev/reference/react-dom/createPortal), used elsewhere in the project for tooltip rendering.

Closes #76 
Co-authored by @mstone121 

~~*Since this is still a work in progress it's assumed there may be some changes to this branch on rebase~~
### Demo

![Screenshot 2023-03-21 at 10 34 00 AM](https://user-images.githubusercontent.com/36374797/226639468-35812216-2c07-4603-99f9-5c003dcc21c8.png)

## Testing Instructions

- `scripts/server`
- Confirm bucket displays over animated map and updates accordingly after selecting "play" and "replay" buttons
- Confirm bucket remains positioned over DC at different window sizes

 ## Checklist

- [x] `fixup!` commits have been squashed
- [x] `CHANGELOG.md` updated with summary of features or fixes, following [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) guidelines
- [x] `README.md` updated if necessary to reflect the changes
- [x] CI passes after rebase

rachelekm committed:
Create spending bucket portal

rachelekm committed:
Add CSS styling

Commit:
Move createPortal into spending bucket element

The useEffect, depending only on the map, prevents the re-render which
motivated moving the div ref up to the map container.

mstone121 reviewed (commented):
This is working well, but passing the ref to the map container and then the child elements feels a little cumbersome. 

I think you can solve the re-render issue with a `useEffect` hook in `AnimatedTotalSpendingBucket`
```js
    const map = useMap();
    const [div, setDiv] = useState<HTMLDivElement | undefined>();

    useEffect(() => {
        const div = document.createElement('div');
        L.popup({ className: 'portal-container', closeButton: false })
            .setLatLng(DC_CENTER)
            .setContent(div)
            .openOn(map);

        setDiv(div);

        return () => {
            setDiv(undefined);
        };
    }, [map]);
```
The key here is the functional form of the setState.

You can also then move the portal inside of `AnimatedTotalSpendingBucket` which helps separating concerns by moving all of the bucket's logic inside itself.

I coded this all out myself to make sure it would work, feel free to borrow these changes: https://github.com/azavea/green-equity-demo/commit/26740a66696933021a7b7893ec2c611a25e0c777.

rachelekm committed:
Wait for spendingAtTimeByState to exist before render

rachelekm committed:
Update CHANGELOG

rachelekm commented:
Thanks for the review and suggestion, that's a really cool fix! I cherry-picked that commit from your branch and added it to this one and updated for co-author credit. Appreciate your help cleaning this up, this is ready for one last look!

mstone121 reviewed (approved):
👍 Looks good!
